Add basic Linux server packaging instructions. (#899)
vircadia/vircadia: INSTALLER.md

1.  Perform a Release build of `package`
      Installer is now available in `build/_CPack_Packages/Darwin/DragNDrop
      
### Linux

#### Server

##### Ubuntu 18.04 | .deb

1. Ensure you are using an Ubuntu 18.04 system, here are the recommended specs:
    ```

>     AWS EC2 Instance Type: C5a.4xlarge

Reviewer: Building an Ubuntu 18.04 server package only needs an amd64/x86_64 machine with at least 40gb of disk space. There is no point in recommending an AWS instance, as it is more expensive than using a virtual machine and takes much longer for a non-AWS user than pretty much anything else. Personally I build in VirtualBox with 6 threads. You can easily build the server on even just one thread.

    ```
    CPU cores to use for Qt5: 16
    ```
14. It will ask you if you would like to proceed with the specified values. If you're happy with the configuration, type `yes`, otherwise enter `no` and press enter to start over. You can press `Ctrl` + `C` simultaneously on your keyboard to exit.
15. Vircadia Builder will now run, it may take a while. See this [table](https://github.com/vircadia/vircadia-builder#how-long-does-it-take) for estimated times.
16. Navigate to the `pkg-scripts` directory.
    ```
    cd ../Vircadia/source/pkg-scripts/
    ```
17. Generate the .rpm package. Set `RPMVERSION` to the same version you entered for the `Release number` on Vircadia Builder. *Advanced users: the version cannot begin with a letter and cannot include underscores or dashes in it.*
    ```
    DEBVERSION="2021.1.0" DEBEMAIL="your-email@somewhere.com" DEBFULLNAME="Your Full Name" ./make-deb-server
    ```
18. If successful, the generated .deb package will be in the `pkg-scripts` folder.

##### Amazon Linux 2 | .rpm

1. Ensure you are using an Amazon Linux 2 system, here are the recommended specs:
    ```
    AWS EC2 Instance Type: C5a.4xlarge

>     CPU Cores: 16

Reviewer: In this case, I feel like recommending a specific instance makes sense as we currently don't have prebuilt Qt packages for Amazon Linux 2 and building Qt takes ages on lower end machines.

    Minimum Disk Space: 40GB

>     ```

Reviewer: ```suggestion
    ```
   You may also use a [virtual machine](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/amazon-linux-2-virtual-machine.html)
```

Keep in mind that you can also use a virtual machine, which is cheaper and probably easier for non-AWS users.
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/amazon-linux-2-virtual-machine.html

    Git repository: https://github.com/digisomni/vircadia/
    ```
8. This value is the tag on the repository. If you would like to use a specific version of Vircadia, typically tags will be named like this: "v2021.1.0-rc".
    ```
    Git tag: master
    # OR, for example
    Git tag: v2021.1.0-rc
    ```
9. This value is the release type. For example, the options are `production`, `pr`, or `dev`. If you are making a build for yourself and others to use then use `production`.
    ```
    Release type: DEV
    # OR, for example we recommend you use
    Release type: PRODUCTION
    ```
10. This value is the release version. Release numbers typically should be in a format of `YEAR-MAJORVERSION-MINORVERSION` which might look like this: `2021.1.0`.
    ```
    Release number: 2021.1.0
    ```
11. This value is the build number. We typically use the hash of the most recent commit on that Git tag which might look like this: `fd6973b`.
    ```
    Build number: fd6973b
    ```
12. This value is the directory that Vircadia will get installed to. You should leave this as the default value unless you are an advanced user.
    ```
    Installation dir: /root/Vircadia
    ```

> 13. This value is the number of CPU cores that the Vircadia Builder will use to compile the Vircadia server. By default it will use all cores available on your build server. You should leave this as the default value it gives you for your build server.

Reviewer: ```suggestion
13. This value is the number of CPU cores that the Vircadia Builder will use to compile the Vircadia server. By default it will use all cores available on your build server given you have enough memory. You should leave this as the default value it gives you for your build server.
```
It is pretty important that people don't just up the amount of threads to all available threads without having enough memory, as the build process might crash and/or the system might freeze.

    ```
    CPU cores to use for Vircadia: 16
    ```

> 14. This value is the number of CPU cores that the Vircadia Builder will use to compile Qt5 (a required component for Vircadia). By default it will use all cores available on your build server. You should leave this as the default value it gives you for your build server.

Reviewer: ```suggestion
14. This value is the number of CPU cores that the Vircadia Builder will use to compile Qt5 (a required component for Vircadia). By default it will use all cores available on your build server given you have enough memory. You should leave this as the default value it gives you for your build server.
```
It is pretty important that people don't just up the amount of threads to all available threads without having enough memory, as the build process might crash and/or the system might freeze.

1.  Perform a Release build of `package`
      Installer is now available in `build/_CPack_Packages/Darwin/DragNDrop
      
### Linux

#### Server

> ##### Ubuntu | .deb

Reviewer: The generated .deb file will only work on Ubuntu 18.04 and potentially higher.
```suggestion
##### Ubuntu 18.04 | .deb
```

1. Ensure you are using an Ubuntu 18.04 system.
2. Get and bootstrap Vircadia Builder.
    ```
    git clone https://github.com/vircadia/vircadia-builder.git
    cd vircadia-builder
    chmod +x vircadia-builder
    ```

> 3. Set up Vircadia Builder to compile the server. See [here](BUILD.md#possible-environment-variables) for possible environment variables and settings.

Reviewer: You are not compiling the server here and building it in the next step.
```suggestion
3. Run Vircadia Builder to build the server. See [here](BUILD.md#possible-environment-variables) for possible environment variables and settings.
```

    ```
    ./vircadia-builder --build server
    ```

> 4. Build the server.

Reviewer: ```suggestion
4. If Vircadia Builder needed to install dependencies and asks you to run it again then do so.
```

    ```
    ./vircadia-builder --build server
    ```
5. Navigate to the `pkg-scripts` directory.
    ```
    cd ../Vircadia/source/pkg-scripts/
    ```
6. Generate the .deb package. Set `DEBVERSION` to a valid semantic version value e.g. "2021.1.0".
    ```

>     DEBVERSION="Semver e.g. 2021.1.0" DEBEMAIL="your-email@somewhere.com" DEBFULLNAME="Your Full Name" ./make-deb-server

Reviewer: A mortal being is likely not going to understand what `Semver` is supposed to mean. Maybe just say `Version`?
```suggestion
    DEBVERSION="Version e.g. 2021.1.0" DEBEMAIL="your-email@somewhere.com" DEBFULLNAME="Your Full Name" ./make-deb-server
```

    ```
7. If successful, the generated .deb package will be in the `pkg-scripts` folder.

##### Amazon Linux 2 | .rpm

1. Ensure you are using an Amazon Linux 2 system.
2. Update the system and install dependencies.
    ```
    sudo yum update -y
    sudo yum install git -y
    sudo yum install rpm-build
    ```
3. Get and bootstrap Vircadia Builder.
    ```
    git clone https://github.com/vircadia/vircadia-builder.git
    cd vircadia-builder
    sudo ./install_amazon_linux_deps.sh

>     chmod +x vircadia-builder

Reviewer: Making vircadia-builder executable is not necessary, as git already does that on by itself. If it was necessary, you wouldn't have been able to run `sudo ./install_amazon_linux_deps.sh` before that.
```suggestion
```

Reviewer 2: This one is a step that Dale outlined for vircadia-builder, is it no longer the case in general?

Reviewer: Depends what you mean with "in general". This is true for git.

    ```

> 4. Set up Vircadia Builder to compile the server. See [here](BUILD.md#possible-environment-variables) for possible environment variables and settings.

Reviewer: Same as above:
You are not compiling the server here and building it in the next step.
```suggestion
4. Run Vircadia Builder to build the server. See [here](BUILD.md#possible-environment-variables) for possible environment variables and settings.
```

    ```
    ./vircadia-builder --build server
    ```

> 5. Build the server.

Reviewer: Same as above:
```suggestion
5. If Vircadia Builder needed to install dependencies and asks you to run it again then do so.
```

    ```
    ./vircadia-builder --build server
    ```
6. Navigate to the `pkg-scripts` directory.
    ```
    cd ../Vircadia/source/pkg-scripts/
    ```
7. Generate the .rpm package. Set `RPMVERSION` to a valid semantic version value e.g. "2021.1.0".
    ```

>     RPMVERSION="Semver e.g. 2021.1.0" ./make-rpm-server

Reviewer: Same suggestion as above:
A mortal being is likely not going to understand what `Semver` is supposed to mean. Maybe just say `Version`?
```suggestion
    RPMVERSION="Version e.g. 2021.1.0" ./make-rpm-server
```
